src/background/initial-visualization-store-data-generator.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { VisualizationConfigurationFactory } from 'common/configs/visualization-configuration-factory';
+import { EnumHelper } from 'common/enum-helper';
+import { DetailsViewPivotType } from 'common/types/store-data/details-view-pivot-type';
+import {
+    TestsEnabledState,
+    VisualizationStoreData,
+} from 'common/types/store-data/visualization-store-data';
+import { VisualizationType } from 'common/types/visualization-type';
+
+export class InitialVisualizationStoreDataGenerator {
+    constructor(
+        private readonly visualizationConfigurationFactory: VisualizationConfigurationFactory,
+    ) {}
+
+    public generateInitialState(
+        persistedData: VisualizationStoreData = null,
+    ): VisualizationStoreData {
+        const defaultTests: TestsEnabledState = {
+            adhoc: {},
+            assessments: {},
+            mediumPass: {},
+        };
+
+        if (this.visualizationConfigurationFactory != null) {
+            EnumHelper.getNumericValues(VisualizationType).forEach((test: VisualizationType) => {
+                const config = this.visualizationConfigurationFactory.getConfiguration(test);
+                defaultTests[config.testMode][config.key] = {
+                    enabled: false,
+                };
+            });
+
+            Object.keys(defaultTests.assessments).forEach(key => {
+                defaultTests.assessments[key].stepStatus = {};
+            });
+
+            Object.keys(defaultTests.mediumPass).forEach(key => {
+                defaultTests.mediumPass[key].stepStatus = {};
+            });
+        }
+        const tests: TestsEnabledState = persistedData?.tests
+            ? { ...defaultTests, ...persistedData.tests }
```

Comment:
This solves the specific issue with the `mediumPass` key being missing, but I worry that it would lead to a similar issue in the future if we ever added a new ndividual test. Perhaps a lodash `merge()` would be a more robust way to solve both this and the issue below with the non-`tests` properties?

Reply:
I think it might be even better to more generally update `persistent-store` to default to merging persisted state into default state instead of keeping persisted state as-is, but that's a scary change to do broadly - maybe best to hold off on that until after the initial MV3 release

Reply:
Discussed offline with @madalynrose and @waabid , decided to use a merge strategy for this store only for now and track doing it more generally after the MV3 release in #6233